packages/core/src/types.ts

```diff
@@ -673,7 +673,11 @@ export interface MachineOptions<TContext, TEvent extends EventObject> {
 export interface MachineConfig<
   TContext,
   TStateSchema extends StateSchema,
-  TEvent extends EventObject
+  TEvent extends EventObject,
+  TActions extends ActionObject<TContext, TEvent> = ActionObject<
+    TContext,
+    TEvent
+  >
```

Comment:
I think this should basically be bound to the action equivalent of `EventObject`. I mean:
```ts
TAction extends { type: string }
```

This would allow for the stuff like this:
https://github.com/davidkpiano/xstate/pull/2266/files#diff-8637c68497a6e188837ab695801648e2f6f72a9831f7420c80409cc354fd1a45R219

but we don't have to embed `TContext` and `TEvent` within that type (or at least I don't see why this would be beneficial right now). Those types are still useful to type actions but it feels like those slots could be just provided as arguments etc, not necessarily as part of the action object~

Reply:
Added `BaseActionObject` for this